sdk/python/feast/infra/materialization/snowflake/snowflake_engine.py

```diff
@@ -0,0 +1,243 @@
+            feature_view, FeatureView
+        ), "Snowflake can only materialize batch features views."
+
+        entities = []
+        for entity_name in feature_view.entities:
+            entities.append(registry.get_entity(entity_name, project))
+
+        (
+            join_key_columns,
+            feature_name_columns,
+            timestamp_field,
+            created_timestamp_column,
+        ) = _get_column_names(feature_view, entities)
+
+        job_id = f"{feature_view.name}-{start_date}-{end_date}"
+
+        try:
+            offline_job = self.offline_store.pull_latest_from_table_or_query(
+                config=self.repo_config,
+                data_source=feature_view.batch_source,
+                join_key_columns=join_key_columns,
+                feature_name_columns=feature_name_columns,
+                timestamp_field=timestamp_field,
+                created_timestamp_column=created_timestamp_column,
+                start_date=start_date,
+                end_date=end_date,
+            )
+
+            table = offline_job.to_arrow()
+            fv_latest_values_sql = offline_job.to_sql()
+
+            """
+            if feature_view.batch_source.field_mapping is not None:
+                fv_latest_values_sql = _run_pyarrow_field_mapping(
+                    table, feature_view.batch_source.field_mapping
+                )
+            """
+
+            join_key_to_value_type = {
+                entity.name: entity.dtype.to_value_type()
+                for entity in feature_view.entity_columns
+            }
+
+            fv_columns = [
+                (field.name, field.dtype.to_value_type())
+                for field in feature_view.features
+            ]
+
+            fv_timestamp = feature_view.batch_source.timestamp_field
+
+            if feature_view.batch_source.created_timestamp_column:
+                fv_created = feature_view.batch_source.created_timestamp_column
+            else:
+                fv_created = None
+
+            online_table = f"""{self.repo_config.online_store.database}"."{self.repo_config.online_store.schema_}"."{self.repo_config.project}_{feature_view.name}"""
+
+            with Session.builder.configs(
+                {"connection": get_snowflake_conn(self.repo_config.offline_store)}
+            ).create() as session:
+
+                print(fv_latest_values_sql)
+                print(online_table)
+                print(fv_columns)
+                print(join_key_to_value_type)
+                print(fv_timestamp)
+                print(fv_created)
+
+                session.call(
+                    "_convert_sql_to_proto()",
+                    fv_latest_values_sql,
+                    online_table,
+                    fv_columns,
+                    join_key_to_value_type,
+                    fv_timestamp,
+                    fv_created,
+                )
+
+            """
+            if self.repo_config.online_store.type == "snowflake.online":
```

Comment:
@adchia @achals for right now I am going to assume that your data is in snowflake / offline store ... and therefore will be 2 code paths ... one to write it to your online store of choice, or one specific to snowflake as online store.

@adchia I guess in theory one could use snowflake ONLY as the materialization engine, but that is probably a pretty expensive endeavor for them if there data doesnt even live in snowflake.